src/hatti/views/map.cljs

```diff
@@ -439,25 +392,58 @@
     (will-receive-props [_ next-props]
       "will-recieve-props resets mapboxglmap
       swiches to geojson source if the map data has changed to geoshapes."
-      (let [{{old-map-data :data} :map-page} (om/get-props owner)
-            {{new-map-data :data} :map-page}  next-props
-            old-field (get-in (om/get-props owner) [:map-page :geofield])
-            new-field (get-in next-props [:map-page :geofield])]
-        (when (or (not= old-field new-field)
-                  (not= (count old-map-data) (count new-map-data))
-                  (not= old-map-data old-map-data))
-          (let [{:keys [flat-form]} (om/get-shared owner)
-                {:keys [mapboxgl-map layer-id geojson]}
-                (om/get-state owner)
-                new-geojson (mu/as-geojson new-map-data flat-form new-field)]
-            (when (and (not-empty new-field) (not= geojson new-geojson))
-              (when (.getLayer mapboxgl-map layer-id)
-                (.removeLayer mapboxgl-map layer-id)
-                (.removeSource mapboxgl-map layer-id))
-              (load-mapboxgl-helper app-state owner
-                                    :geojson new-geojson
-                                    :geofield new-field)
-              (put! shared/event-chan {:data-updated true}))))))))
+      (let [{{old-map-data :data
+              old-field :geofield
+              {old-cell-width :cell-width} :hexbins
+              old-viewby :view-by
+              old-style :style} :map-page} (om/get-props owner)
+            {{new-map-data :data
+              new-field :geofield
+              {show-hexbins? :show?
+               new-cell-width :cell-width
+               hide-points? :hide-points?} :hexbins
+              new-viewby :view-by
+              new-style :style} :map-page} next-props
+            {:keys [mapboxgl-map layer-id geojson]} (om/get-state owner)
+            {:keys [flat-form]} (om/get-shared owner)
+            data-changed? (or (not= old-field new-field)
+                              (not= (count old-map-data) (count new-map-data))
+                              (not= old-map-data old-map-data))
+            new-geojson (if data-changed?
+                          (mu/as-geojson new-map-data flat-form new-field)
+                          geojson)
+            view-by-changed? (and (not= old-viewby new-viewby)
+                                  (not (nil? old-viewby)))
+            cell-width-changed? (not= old-cell-width new-cell-width)
+            opts (when (and show-hexbins? view-by-changed?)
+                   (vb/get-selected-ids new-viewby))
+            hexbin-opts (assoc opts :cell-width new-cell-width)]
+        ;; Update layers if data changes
+        (when (and data-changed? (not-empty new-field)
+                   (not= geojson new-geojson))
+          (when (.getLayer mapboxgl-map layer-id)
```

Comment:
Isn't this the same as https://github.com/onaio/hatti/pull/174/files#diff-48c0c62913167da8085340a83a0734d6R591, helper function?